### PR TITLE
stdalgos (7/10): team-level API, and rst update for release 4.2

### PR DESCRIPTION
**Pull request by antoinemeyer5**
Add team-level API to and convert to rst the following algorithms:
```
Kokkos_Unique.hpp
Kokkos_UniqueCopy.hpp
Kokkos_Rotate.hpp
Kokkos_RotateCopy.hpp
Kokkos_Remove.hpp
Kokkos_RemoveIf.hpp
```

NOTE: this follows the new structure for documenting the API, but does not use sphinx because the lack of properly aligning things make things quite unreadable.

Open individual html with Firefox (must be firefox or it won't display correctly) 
[Archive.zip](https://github.com/kokkos/kokkos-core-wiki/files/13653065/Archive.zip)

### CODE REVIEW

File: docs/source/API/algorithms/std-algorithms/all/StdRemoveIf.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                  UnaryPredicate pred);
+
+Parameters and Requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. |remove| replace:: ``remove``
+.. _remove: ./StdRemove.html
+
+- ``exespace``, ``first``, ``last``, ``view``: same as in |remove|_
+
```
Comment (antoinemeyer5) [inline docs/source/API/algorithms/std-algorithms/all/StdRemoveIf.rst:74]: miss teamHandle description

**Review by antoinemeyer5 — COMMENTED** [review]
add it @antoinemeyer5

File: docs/source/API/algorithms/std-algorithms/all/StdRemoveIf.rst (hunk 1 above)
Comment (antoinemeyer5) [reply docs/source/API/algorithms/std-algorithms/all/StdRemoveIf.rst:74]: fixed

**Review by ajpowelsnl — APPROVED** [review]
@antoinemeyer5 , @fnrizzi , @crtrott -- a quick scan says it's good to go

**Pull request merged**